services/ui-src/src/components/menus/Sidebar.tsx

```diff
@@ -1,190 +1,250 @@
-  const toggleVisibility = (parentName: string) => {
-    if (openMenuItemList.includes(parentName)) {
-      setOpenMenuItems(openMenuItemList.filter((name) => name !== parentName));
-    } else {
-      setOpenMenuItems((state) => [...state, parentName]);
-    }
-  };
+  const { pathname } = useLocation();
+  const isMcparReport = pathname.includes("/mcpar");
+  const [isOpen, setIsOpen] = useState(true);
   return (
-    <Box sx={sx.root}>
-      <Flex sx={sx.sideNavTopFlex}>
-        <Text sx={sx.sideNavHeader}>MCPAR Report Submission Form</Text>
-        <CloseButton />
-      </Flex>
-      {LinkItems.map((link) => (
-        <NavItem
-          key={link.name}
-          path={link.path}
-          hasSubItems={!!link.hasSubItems}
-          name={link.name}
-          hidden={!!link.parent && !openMenuItemList.includes(link.parent)}
-          onClick={() => {
-            toggleVisibility(link.name);
-          }}
-        />
-      ))}
-    </Box>
+    <>
+      {isMcparReport && (
+        <Box sx={sx.root} className={isOpen ? "open" : "closed"}>
+          <Box
+            as="button"
+            sx={sx.closeButton}
+            onClick={() => setIsOpen(!isOpen)}
+          >
+            <ArrowIcon title="closeNavBarButton" direction="left" />
+          </Box>
+          <Box sx={sx.topBox}>
+            <Heading sx={sx.title}>MCPAR Report Submission Form</Heading>
+          </Box>
+          {LinkItems.map((section) => (
+            <NavSection
+              key={section.name}
+              section={section}
+              level={1}
+              basePath="/mcpar"
+            />
+          ))}
+        </Box>
+      )}
+    </>
   );
 };
 
-const CloseButton = () => {
+interface NavSectionProps {
+  key: string;
+  section: LinkItemProps;
+  level: number;
+  basePath: string;
+}
+
+const NavSection = ({ section, level, basePath }: NavSectionProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { name, path, children } = section;
+  const itemPath = `${basePath}${path}`;
   return (
-    <Flex align="center" paddingY="0.5rem" marginLeft="1.25rem" sx={sx.temp}>
-      <ArrowIcon title="closeNavBarButton" direction="left" />
-    </Flex>
+    <>
+      <React.Fragment key={itemPath}>
+        {children ? (
+          <Box as="button" onClick={() => setIsOpen(!isOpen)}>
+            <NavItem name={name} level={level} hasChildren={!!children} />
+          </Box>
+        ) : (
+          <>
+            <Link as={RouterLink} to={itemPath}>
+              <NavItem name={name} level={level} hasChildren={!!children} />
+            </Link>
+          </>
+        )}
+        {!!children && (
+          <Collapse in={isOpen}>
+            {children.map((section) => (
+              <NavSection
+                key={section.name}
+                section={section}
+                level={level + 1}
+                basePath={itemPath}
+              />
+            ))}
+          </Collapse>
+        )}
```

Comment:
Here we get fancy calling the component from within itself, but only if it has children. The collapse makes the animation nice.